src/averageAges.js

```diff
@@ -37,7 +47,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const arrayOfWomen = withChildren
+    ? people.filter(person => person.sex === 'f'
+    && people.some(child => child.mother === person.name))
+    : people.filter(person => person.sex === 'f');
+
+  return arrayOfWomen.reduce(function(average, woman) {
```

Comment:
Consider using more descriptive variable names, like `womenAverageAge` instead of `average`. 🤔